app/services/course/assessment/submission/statistics_download_service.rb

```diff
@@ -0,0 +1,143 @@
+      end
+    end
+
+    output_file
+  end
+
+  def download_statistics_header(csv)
+    csv << [I18n.t('course.assessment.submission.submissions.statistics_download_service.name'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.status'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.grade'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.max_grade'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.exp_points'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.start_date_time'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.submitted_date_time'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.time_taken'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.graded_date_time'),
+            I18n.t('course.assessment.submission.submissions.statistics_download_service.grading_time')]
+  end
+
+  def download_statistics(csv, submission)
+    csv << [submission.course_user.name,
+            submission.workflow_state,
+            submission.grade,
+            submission.maximum_grade,
+            csv_exp_points(submission),
+            csv_created_at(submission),
+            csv_submitted_date_time(submission),
+            csv_time_taken(submission),
+            csv_graded_at(submission),
+            csv_grading_time(submission)]
+  end
+
+  def csv_empty
+    I18n.t('course.assessment.submission.submissions.statistics_download_service.csv_empty')
+  end
+
+  def csv_time_taken(submission)
+    if submission.submitted_at && submission.created_at
+      ActionController::Base.helpers.
+        distance_of_time_in_words(submission.submitted_at - submission.created_at)
+    else
+      csv_empty
+    end
+  end
+
+  def csv_exp_points(submission)
+    if submission.current_points_awarded
+      submission.current_points_awarded
+    else
+      csv_empty
+    end
+  end
+
+  def csv_created_at(submission)
+    if submission.created_at
+      format_datetime(submission.created_at, :long, user: @course_user.user)
+    else
+      csv_empty
+    end
+  end
+
+  def csv_submitted_date_time(submission)
+    if submission.submitted_at
+      format_datetime(submission.submitted_at, :long, user: @course_user.user)
+    else
+      csv_empty
+    end
+  end
+
+  def csv_graded_at(submission)
+    if submission.last_graded_at
+      format_datetime(submission.last_graded_at, :long, user: @course_user.user)
+    else
+      csv_empty
+    end
+  end
+
+  def csv_grading_time(submission)
+    if submission.last_graded_at && submission.submitted_at
+      ActionController::Base.
+        helpers.distance_of_time_in_words(submission.last_graded_at - submission.submitted_at)
```

Comment:
I am not sure why this field is in human readable format, I suspect that is hard to parse in CSV.  Are there anything that I am missing?

Reply:
Let's use minutes instead?

Reply:
how about `HH:MM:SS`? any specific requirement for this?

Reply:
Nope. We have another column called grading date/time. This column is for the time period for grading.

Reply:
How about https://en.wikipedia.org/wiki/ISO_8601

Duration will be sth like`3H2M5S`

Reply:
OK.